metadata_service/proxy/atlas_proxy.py

```diff
@@ -182,11 +179,12 @@ def _create_reader(self, metadata_guid: str, user_guid: str, reader_qn: str, tab
             'attributes': {'qualifiedName': reader_qn,
                            'isFollowing': True,
                            'count': 0,
-                           'entityMetadata': {'guid': metadata_guid},
-                           'user': {'guid': user_guid},
-                           'entityUri': table_uri}
+                           'user': {'guid': user_guid}}
         }
-        self._driver.entity_bulk.create(data={'entities': [reader_entity]})
+
+        readers = table_entity.entity[self.ATTRS_KEY]['readers'].append(reader_entity)
+        table_entity.entity[self.ATTRS_KEY]['readers'] = readers
+        table_entity.update()
```

Comment:
Does this create a new reader entity as well ? (just curious is you need to create a reader entity first and then update in table readers.

Reply:
You are right, encounter this problem while generating the test data - reader entity must be first created then appended. Fixed in the latest commit.